app/assets/javascripts/components/admin/matched_lessons.es6.jsx

```diff
@@ -0,0 +1,232 @@
+        </tbody>
+      </table>
+    )
+  }
+
+  renderUpcomingTable() {
+    const { upcomingLessons } = this.state;
+    let items;
+
+    if (upcomingLessons) {
+      items = upcomingLessons.map((lesson, key) => {
+        return (
+          <MatchingUpcomingLessonRow lesson={lesson} key={key}
+            fetch={this.fetchUpcomingLessons}/>
+        )
+      })
+    }
+
+    return (
+      <table className="interactive matched-lesson-table">
+        <thead id="table-head">
+          <tr>
+            <th>Day</th>
+            <th>Time</th>
+            <th>Price</th>
+            <th>Options</th>
+          </tr>
+        </thead>
+        <tbody>
+          { items }
+        </tbody>
+      </table>
+    )
+  }
+
+  renderSummary() {
+    const { student, teacher } = this.props;
+
+    let studentName = `${student.first_name} ${student.last_name}`;
+    let teacherName = `${teacher.first_name} ${teacher.last_name}`;
+
+    return (
+      <div className="matching-summary">
+        <div className="matching-summary-item">
+          <div className="label">Student</div>
+          <h4 className="info">{studentName}</h4>
+        </div>
+        <div className="matching-summary-item">
+          <div className="label">Teacher</div>
+          <h4 className="info">{teacherName}</h4>
+        </div>
+        <div className="matching-summary-item">
+          <div className="label">Instrument</div>
+          <h4 className="info">{this.props.matching.instrument}</h4>
+        </div>
+      </div>
+    )
+  }
+
+  render() {
+    let lessons, tableView;
+    let  { tab } = this.state;
+
+    if (tab == "past") {
+      tableView = this.renderPastTable();
+    } else {
+      tableView = this.renderUpcomingTable();
+    }
+
+    return (
+      <div className="page-wrapper">
+        <AdminHeader />
+        <div className="content-wrapper matched-lessons-page">
+          <h1>Matching Lessons</h1>
+          {this.renderSummary()}
+          <div>
+            <ButtonGroup className="matched-lessons-tab">
+              <Button onClick={this.switchTab.bind(this, 'past')}
+                className={tab == "past" ? "active" : ""}>
+                Past Lessons
+              </Button>
+              <Button onClick={this.switchTab.bind(this, 'upcoming')}
+                className={tab == "upcoming" ? "active" : ""}>
+                Upcoming Lessons
+              </Button>
+            </ButtonGroup>
+            { tableView }
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+class MatchingPastLessonRow extends React.Component {
+
+  render() {
+    const { lesson } = this.props;
+    const start = moment(lesson.start_time);
+
+    return (
+      <tr>
+        <td>{start.format("LL")}</td>
+        <td>{start.format("h:mm a")}</td>
+        <td>{lesson.is_paid ? "Yes" : "No"}</td>
+        <td>${lesson.price}</td>
+      </tr>
+    );
+  }
+}
+
+class MatchingUpcomingLessonRow extends React.Component {
```

Comment:
oh looks like you have the cancel modal, but could you add the rescheduling part of it? (should be very similar to the reschedule_modal we have)